### PR TITLE
Add users authentication

### PR DESCRIPTION
**Pull request by SHtress**
Changes from #7.

### CODE REVIEW

File: sc-kpm/sc-python/services/http_api/auth/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__all__=['config','validators', 'constants']
```
Comment (AlexandrZagorskiy) [inline sc-kpm/sc-python/services/http_api/auth/__init__.py:1]: space

File: sc-kpm/sc-python/services/http_api/auth/config.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import http_api.auth.constants as cnt
+from pathlib import Path
+
+params = {
+    # auth params
+    cnt.PUBLIC_KEY_PATH: Path(__file__).parent / 'public.pem' ,
+    cnt.AUTH_FLAG: False,
```
Comment (AlexandrZagorskiy) [inline sc-kpm/sc-python/services/http_api/auth/config.py:7]: remove it

File: sc-kpm/sc-python/services/http_api/auth/constants.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+ISSUER = 'issuer'
+PUBLIC_KEY_PATH = 'public_key'
+AUTH_FLAG = 'auth_flag'
```
Comment (AlexandrZagorskiy) [inline sc-kpm/sc-python/services/http_api/auth/constants.py:3]: remove it too

File: sc-kpm/sc-python/services/http_api/http_api.py [hunk 4]
```diff
@@ -13,6 +14,18 @@
 
 self_path = os.path.dirname(__file__)
 
+def check_auth():
+    path = os.path.join(Path(__file__).parent.parent.parent.parent.parent, "CMakeLists.txt")
+    try:
+        with open(path, 'r') as file:
+            if "option(AUTH_ENABLED ON)" in file.read():
```
Comment (AlexandrZagorskiy) [inline sc-kpm/sc-python/services/http_api/http_api.py:21]: move to config

File: .gitignore [hunk 5]
```diff
@@ -19,4 +19,6 @@ manifest.json
 __pycache__
 .DS_store
 .antlr
-.cache
```
Comment (AlexandrZagorskiy) [inline .gitignore:22]: return it

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

**Review by AlexandrZagorskiy — APPROVED** [review]

File: sc-kpm/sc-python/services/http_api/http_api.py [hunk 6]
```diff
@@ -4,15 +4,29 @@
...
 from keynodes import Keynodes
+from http_api.auth.config import params
+import http_api.auth.constants as cnt
 
 from sc import *
 
 self_path = os.path.dirname(__file__)
 
+def check_auth():
+    path = os.path.join(Path(__file__).parent.parent.parent.parent.parent, "CMakeLists.txt")
```
Comment (NikitaZotov) [inline sc-kpm/sc-python/services/http_api/http_api.py:20]: Can we avoid relative path scheme using in code?

File: sc-kpm/sc-python/services/http_api/http_api.py [hunk 7]
```diff
@@ -4,15 +4,29 @@
...
+    try:
+        with open(path, 'r') as file:
+            if params[cnt.AUTH_FLAG_STRING] in file.read():
+                return True
+            else:
+                return False
```
Comment (NikitaZotov) [inline sc-kpm/sc-python/services/http_api/http_api.py:26]:
```suggestion
            return params[cnt.AUTH_FLAG_STRING] in file.read():
```

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

**Comment by KovalM** [comment]
Authentication will be rewrited for new sc-server implementation from #62

**Pull request closed without merging**